core/src/commonMain/kotlin/org/aquila3d/core/math/Matrix4.kt

```diff
@@ -0,0 +1,104 @@
+package org.aquila3d.core.math
+
+/**
+ * 4x4 column major matrix.
+ *
+ * | 0 4  8 12 |
+ * | 1 5  9 13 |
+ * | 2 6 10 14 |
+ * | 3 7 11 15 |
+ */
+open class Matrix4 {
+
+    /**
+     * Storing data this way allows us to index easier. values[0][3] represents index 3 shown above while values[3][0]
+     * represents index 12.
+     */
+    protected val values: Array<Column>
+
+    /**
+     * Creates a new [Matrix4] initialized to the Identity matrix.
+     */
+    constructor() {
+        values = Array(4) { Column(DoubleArray(4) { 0.0 }) }
+        values[0][0] = 1.0
+        values[1][1] = 1.0
+        values[2][2] = 1.0
+        values[3][3] = 1.0
+    }
+
+    constructor(other: Matrix4) {
+        values = other.values
```

Comment:
Will this be a dangerous op in the future if future methods modify the internal state?

Reply:
Dangerous because its a shallow copy? Yes I would say so. I will make it clone the values.